tests/functional/utils/chart_certification.py

```diff
@@ -51,8 +51,11 @@ def set_git_username_email(self, repo, username, email):
         username (str): git username to set
         email (str): git email to set
         """
-        repo.config_writer().set_value("user", "name", username).release()
-        repo.config_writer().set_value("user", "email", email).release()
+        git_path = repo.git.rev_parse("--show-toplevel")
+        # make sure only one process is modifying the git config
+        if not pathlib.Path(f'{git_path}/.git/config.lock').exists():
```

Comment:
I think even with this check there is a possibility that two pytest thread can check this condition at the same time and for both the thread this condition is true and try to execute the next portion of code. 
One approach I can think of is implement the similar mechanism as config_writer() using to synchronize the update of config, implement some .lock file to synchronize this portion of code.

Reply:
The `config.lock` file was created by git functioning as a mutex lock, and I agree it might not be atomic. Don't think we can add a lock _after_ the contention point because we will be already failing at that point. Did you encounter failure case for this PR?

Reply:
yes I encountered the same error when running with this change as well.